com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/modelinference/InferredModelAssociator.java

```diff
@@ -64,8 +64,18 @@
    * An adapter that holds the mapping between source- and inferred-model elements.
    */
   public static class Adapter extends AdapterImpl {
-    private final Map<EObject, Deque<EObject>> sourceToInferredModelMap = newLinkedHashMapWithCapacity(40);
-    private final Map<EObject, Deque<EObject>> inferredModelToSourceMap = newLinkedHashMapWithCapacity(40);
+    private Map<EObject, Deque<EObject>> sourceToInferredModelMap;
+    private Map<EObject, Deque<EObject>> inferredModelToSourceMap;
+    private static final int CAPACITY = 40;
+
+    public Adapter() {
+      this(CAPACITY);
+    }
+
+    protected Adapter(final int capacity) {
+      sourceToInferredModelMap = newLinkedHashMapWithCapacity(capacity);

```

Comment:
would it make sense to use Collections.emptyMap if the capacity is 0?

Reply:
done